### ISSUE TITLE
Error when validating XML with namespace prefixes

### ISSUE THREAD

**Issue by MatteBailey** [tlambert03/ome-types#89]
Hello!

I was attempting to validate the XML attached below, and got this error:
```
Traceback (most recent call last):
  File "/home/matteb/code/czi-to-ome-xslt/ome_types_test.py", line 49, in <module>
    from_xml(str(ome))
  File "/home/matteb/virtualenvs/czi-to-ome-xslt/lib/python3.9/site-packages/ome_types/__init__.py", line 42, in from_xml
    return OME(**d)  # type: ignore
TypeError: __init__() got an unexpected keyword argument 'ome:Experimenter'
```

From what I can gather, `ome-types`  expects to find `Experimenter`, but is stumbling on the namespace prefix in `ome:Experimenter`. However, using namespace prefixes like this is valid XML, and I have validated this file against the OME schema using several other methods (`xmlschema`, `lxml`, and [`xmlvalid`](https://docs.openmicroscopy.org/bio-formats/6.6.1/users/comlinetools/xml-validation.html)).

Anything I'm missing here?
[produced.ome.zip](https://github.com/tlambert03/ome-types/files/6588006/produced.ome.zip)

**Comment by tlambert03** [comment]
thanks @MatteBailey and thank you for the example.
that's a bug...

you're correct, it's getting validated and parsed correctly by xmlschema, but we need to strip the global namespace in this case before passing that parsed dict to `OME()` ... will patch soon.

**Comment by tlambert03** [comment]
@MatteBailey, one more question regarding that specific XML example...
The `ome:AcquisitionDate` is `2019-06-27T18:39:25.8078869Z` (7 digits of subseconds)... which `xmlschema` has an issue with:
```pytb
            elif (kwargs['microsecond'] // 1 ** zeros) > 999999:
                msg = "Invalid value {} for microsecond"
>               raise OverflowError(msg.format(kwargs['microsecond']))
E               OverflowError: Invalid value 8078869 for microsecond
```

and from a little googling, I can't see many examples of the ISO 8601 standard showing more than 6 digits.  

I can fix the namespace issue pretty easily... my preference would be to not special-case the digits (which means you'd have to catch it on your end).  that ok with you?

**Comment by MatteBailey** [comment]
@tlambert03 Interestingly, I can't reproduce the error you're seeing with `xmlschema`. Below is what I'm running with `xmlschema` 1.4.1, let me know if there's something different I should try.
```python
from xmlschema import XMLSchema

if __name__ == '__main__':
    ome_schema = XMLSchema('ome.xsd')
    ome_schema.validate('produced.ome.xml')

```

Regardless, I don't think we necessarily need support for seven digits of subseconds, I believe it's just what was included in the file from the microscope.

**Comment by tlambert03** [comment]
yeah, strange:  here's my full output if you're curious (I know it's peripheral to the main point here... sorry for the delay in the fix,  busy week)

<details>

```python
In [1]: from xmlschema import XMLSchema, __version__

In [2]: print(__version__)
1.4.1

In [3]: schema = XMLSchema('https://www.openmicroscopy.org/Schemas/OME/2016-06/ome.xsd')

In [4]: schema.validate('/Users/talley/Downloads/produced.ome.xml')
---------------------------------------------------------------------------
OverflowError                             Traceback (most recent call last)
<ipython-input-4-a9d24c0e93bc> in <module>
----> 1 schema.validate('/Users/talley/Downloads/produced.ome.xml')

~/miniconda3/envs/ome/lib/python3.8/site-packages/xmlschema/validators/schema.py in validate(self, source, path, schema_path, use_defaults, namespaces)
   1410         :raises: :exc:`XMLSchemaValidationError` if XML *data* instance is not a valid.
   1411         """
-> 1412         for error in self.iter_errors(source, path, schema_path, use_defaults, namespaces):
   1413             raise error
   1414

~/miniconda3/envs/ome/lib/python3.8/site-packages/xmlschema/validators/schema.py in iter_errors(self, source, path, schema_path, use_defaults, namespaces)
   1509                     return
   1510
-> 1511             for result in xsd_element.iter_decode(elem, **kwargs):
   1512                 if isinstance(result, XMLSchemaValidationError):
   1513                     yield result

~/miniconda3/envs/ome/lib/python3.8/site-packages/xmlschema/validators/elements.py in iter_decode(self, elem, validation, **kwargs)
    620                     yield self.validation_error(validation, error, **kwargs)
    621
--> 622             for result in content_decoder.iter_decode(elem, validation, **kwargs):
    623                 if isinstance(result, XMLSchemaValidationError):
    624                     yield self.validation_error(validation, result, elem, **kwargs)

~/miniconda3/envs/ome/lib/python3.8/site-packages/xmlschema/validators/groups.py in iter_decode(self, elem, validation, **kwargs)
    755                 continue
    756
--> 757             for result in xsd_element.iter_decode(child, validation, **kwargs):
    758                 if isinstance(result, XMLSchemaValidationError):
    759                     yield result

~/miniconda3/envs/ome/lib/python3.8/site-packages/xmlschema/validators/elements.py in iter_decode(self, elem, validation, **kwargs)
    620                     yield self.validation_error(validation, error, **kwargs)
    621
--> 622             for result in content_decoder.iter_decode(elem, validation, **kwargs):
    623                 if isinstance(result, XMLSchemaValidationError):
    624                     yield self.validation_error(validation, result, elem, **kwargs)

~/miniconda3/envs/ome/lib/python3.8/site-packages/xmlschema/validators/groups.py in iter_decode(self, elem, validation, **kwargs)
    755                 continue
    756
--> 757             for result in xsd_element.iter_decode(child, validation, **kwargs):
    758                 if isinstance(result, XMLSchemaValidationError):
    759                     yield result

~/miniconda3/envs/ome/lib/python3.8/site-packages/xmlschema/validators/elements.py in iter_decode(self, elem, validation, **kwargs)
    679                             value = kwargs['filler'](self)
    680             else:
--> 681                 for result in content_decoder.iter_decode(text, validation, **kwargs):
    682                     if isinstance(result, XMLSchemaValidationError):
    683                         yield self.validation_error(validation, result, elem, **kwargs)

~/miniconda3/envs/ome/lib/python3.8/site-packages/xmlschema/validators/simple_types.py in iter_decode(self, obj, validation, **kwargs)
    595
    596         try:
--> 597             result = self.to_python(obj)
    598         except (ValueError, DecimalException) as err:
    599             yield XMLSchemaDecodeError(self, obj, self.to_python, reason=str(err))

~/miniconda3/envs/ome/lib/python3.8/site-packages/elementpath/datatypes/datetime.py in fromstring(cls, datetime_string, tzinfo)
    266             elif (kwargs['microsecond'] // 1 ** zeros) > 999999:
    267                 msg = "Invalid value {} for microsecond"
--> 268                 raise OverflowError(msg.format(kwargs['microsecond']))
    269             else:
    270                 kwargs['microsecond'] = int(match.groupdict()['microsecond'][:6])

OverflowError: Invalid value 8078869 for microsecond
```

</details>

**Comment by tlambert03** [comment]
should be fixed in 0.2.5, which is now released.  I added this xml to the tests as well (with a modified acquisition date, which we can tackle later if you want).  thanks for reporting!
